src/ofxstatement/parser.py

```diff
@@ -46,7 +46,7 @@ def parse(self) -> Statement:
             stmt_line = self.parse_record(line)
             if stmt_line:
                 stmt_line.assert_valid()
-                self.statement.lines.append(stmt_line)
+                self.statement.lines.append(stmt_line)  # type: ignore
```

Comment:
This `# type: ignore` is not needed now?

FWICT,  `Statement` class has two lists:

```python

class Statement(Printable):
    lines: List["StatementLine"]
    invest_lines: List["InvestStatementLine"]
```

Plugins that parse investment statement are supposed to add lines to `invest_lines`. Those that parse regular bank statemrents add to `lines`.   

Someone who added `InvestStatementLine` feature probably just overrode `parse()` and didn't use this implementation at all. And we didn't bother to add the same level of support for the investment statements that regular statements have.

That said, I think `parse_invest_record` is the move in right direction, we just need to use it here somehow, because right now it looks like a lonely unused method. What I suggest is to return None by default from both `parse_record()` and `parse_invest_record()`, indicating they wasn't able to parse the record.

Then, in `parse` itself add another couple of lines:

```python:
if stmt_line:
    ...
else:
    invest_stmt_line = self.parse_invest_record(line)
    if invest_stmt_line:
        self.statement.invest_lines.append(invest_stmt_line)
```

This way users can override one of the `parse_*_record` depending on what statements their plugins are parsing and still make use of the generic `parse()` method.

And we won't need these shameful  `# type: ignore` comments as well :)

What do you think?